src/models/room.ts

```diff
@@ -1981,8 +1981,12 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         timelineSet: Optional<EventTimelineSet>,
         thread: Thread,
         toStartOfTimeline: boolean,
+        recreateEvent: boolean,
     ): void => {
         if (timelineSet && thread.rootEvent) {
+            if (recreateEvent) {
+                timelineSet.removeEvent(thread.id);
+            }
```

Comment:
could this not be made unconditional? As I understand it, the only time that `updateThreadRootEvent` is called with `recreateEvent=False`, the event will not be in the timelineset anyway.

Reply:
`createThread` is also called for every thread we load in the thread list. That means when scrolling through the thread list, it'll get called repeatedly, with old threads. In that case we want to avoid moving the event to the bottom of the list.